### PR TITLE
IGNITE-8666 Add ability of filtering data during datasets creation

### PR DESCRIPTION
**Pull request by dmitrievanthony**
…asetBuilder.

### CODE REVIEW

File: modules/ml/src/main/java/org/apache/ignite/ml/dataset/impl/cache/CacheBasedDatasetBuilder.java [hunk 1]
```diff
@@ -52,15 +53,31 @@
...
+    }
+
+    /**
+     * Constructs a new instance of cache based dataset builder that makes {@link CacheBasedDataset}.
+     *
+     * @param ignite Ignite instance.
+     * @param upstreamCache Ignite Cache with {@code upstream} data.
+     * @param pred Predicate that filters {@code upstream} data.
+     */
+    public CacheBasedDatasetBuilder(Ignite ignite, IgniteCache<K, V> upstreamCache, IgniteBiPredicate<K, V> pred) {
```
Comment (zaleslaw) [inline modules/ml/src/main/java/org/apache/ignite/ml/dataset/impl/cache/CacheBasedDatasetBuilder.java]: pred is strange unreadable name

File: modules/ml/src/main/java/org/apache/ignite/ml/dataset/impl/cache/util/ComputeUtils.java [hunk 2]
```diff
@@ -253,4 +273,38 @@
         IgniteCache<Integer, C> datasetCache = ignite.cache(datasetCacheName);
         datasetCache.put(part, ctx);
     }
+
+    /**
+     * Computes number of entries selected from the cache ny the query.
```
Comment (zaleslaw) [inline modules/ml/src/main/java/org/apache/ignite/ml/dataset/impl/cache/util/ComputeUtils.java]: ny? fix please

File: modules/ml/src/main/java/org/apache/ignite/ml/dataset/impl/cache/CacheBasedDatasetBuilder.java (hunk 1 above)
Comment (dmitrievanthony) [reply modules/ml/src/main/java/org/apache/ignite/ml/dataset/impl/cache/CacheBasedDatasetBuilder.java]: Renamed `pred` -> `filter`.

File: modules/ml/src/main/java/org/apache/ignite/ml/dataset/impl/cache/util/ComputeUtils.java (hunk 2 above)
Comment (dmitrievanthony) [reply modules/ml/src/main/java/org/apache/ignite/ml/dataset/impl/cache/util/ComputeUtils.java]: Fixed.

**Pull request closed without merging**